### PR TITLE
Support not_changeset_full_match for beats

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Support a new when condition to full match the changeset

## Why is it important?

Otherwise, there is a chance that some PRs won't trigger mandatory fields since it uses a partial match at the moment.

## For the review

- First commit (https://github.com/elastic/apm-pipeline-library/pull/890/commits/e1ba36d22b1b1dc09b5a0b795becf2f7f9b11db4) was a refactor, so almost all the logic for the `whenChangeset` was moved to a new function `changeset`.
- Second commit (https://github.com/elastic/apm-pipeline-library/pull/890/commits/578f4f80b197a9d6a9b21098ecb3c3e5f2d9d461), added the new when condition.

### CODE REVIEW

File: vars/beatsWhen.groovy [hunk 1]
```diff
@@ -59,63 +60,81 @@ private Boolean whenBranches(Map args = [:]) {
...
 private Boolean whenChangeset(Map args = [:]) {
+  def name = 'Changeset'
   if (args.content?.get('changeset')) {
-    // Gather macro changeset entries
-    def macro = [:]
-    args?.changeset?.each { k,v ->
-      macro[k] = v
-    }
+    def arguments = args
+    arguments.name = name
```
Comment (cachedout) [inline vars/beatsWhen.groovy:66]: I'm not understanding the need for these two lines. Am I just not seeing where `arguments` is used?

**Review by cachedout — APPROVED** [review]

File: vars/beatsWhen.groovy (hunk 1 above)
Comment (v1v) [reply vars/beatsWhen.groovy:66]: aha! good catch

File: vars/beatsWhen.groovy [hunk 2]
```diff
@@ -59,63 +60,81 @@ private Boolean whenBranches(Map args = [:]) {
...
+    return changeset(args)
```
Comment (v1v) [inline vars/beatsWhen.groovy:67]:
```suggestion
    return changeset(arguments)
```

File: vars/beatsWhen.groovy (hunk 1 above)
Comment (v1v) [reply vars/beatsWhen.groovy:66]: I've just fixed below, but I'll add some UTs now

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**